### ISSUE TITLE
OB-6/Prophet-6(?) not detected when its MIDI channel is set to MPE

### ISSUE THREAD

**Issue by Florent-H** [christofmuc/KnobKraft-orm#117]
The latest firmware for the OB-6 enables MPE (MIDI Polyphonic Expression) when the global setting for MIDI channel is set to MPE. However, in MPE mode, my OB-6 is not detected by KnobKraft Orm. On the other hand, It is detectable when setting the MIDI channel to any individual channel of the 16 available.

To note, the OB-6's 6 voices operate on channels 2-7 in MPE mode.

Could you please implement MPE for the OB-6 (and Prophet 6 too I imagine) in KnobKraft Orb?

Thank you so much!

**Comment by Florent-H** [comment]
I am closing this Issue, since it appears rather that my MIDI devices are faulty. MPE works on some devices but not others.

**Comment by Florent-H** [comment]
Sorry for the back and forth, but I'm curious as to why the MIDI channel on the OB-6 is automatically set to ALL when I auto-detect with the MIDI channel originally set to MPE.

Does the ALL MIDI channel still work with MPE? Is it required by KnobKraft Orm?

**Comment by christofmuc** [comment]
Hi Florian, thanks for the info. I presume you are looking at the global settings page of the OB-6, or are you looking at the log window? I remember that the OB-6 had quite a few sysex bugs in some firmware versions regarding the global settings, I had even filed them with Sequential and they promised to fix them. It would be good if that happened now.

Can you give me the firmware version number of your OB-6, and maybe paste the auto detection MIDI log here so I can look at the data going back and forth? I have updated my OB-6 a while ago, but never used the MPE mode, so probably I just haven't tested that at all.

Cheers and thanks!

**Comment by Florent-H** [comment]
Hi Christof,

No worries with the name! Florian is more common than Florent, so I understand why you would accidentally swap them =)

I am indeed looking at the log window. Here's the log when trying to auto-configure my OB-6 with MPE as the MIDI channel.

```
11:03:27.531: Out UM-ONE Sysex [f0 01 2e 0e f7]
11:03:27.552: In  UM-ONE Sysex [f0 01 2e 0f 0c 32 11 02 00 02 02 01 00 00 00 00 02 00 00 00 00 00 00 00 01 2f 00 01 01 f7]
11:03:27.632: Out UM-ONE Sysex [f0 01 2e 0e f7]
11:03:33.654: Out Microsoft GS Wavetable Synth Sysex [f0 01 2e 0e f7]
11:03:33.757: Out SSL V-MIDI Port 1 Sysex [f0 01 2e 0e f7]
11:03:33.860: Out SSL V-MIDI Port 2 Sysex [f0 01 2e 0e f7]
11:03:33.963: Out SSL V-MIDI Port 3 Sysex [f0 01 2e 0e f7]
11:03:34.067: Out SSL V-MIDI Port 4 Sysex [f0 01 2e 0e f7]
11:03:34.170: Out SSL V-MIDI Port 5 Sysex [f0 01 2e 0e f7]
11:03:34.273: Out SSL V-MIDI Port 6 Sysex [f0 01 2e 0e f7]
11:03:34.376: Out SSL V-MIDI Port 7 Sysex [f0 01 2e 0e f7]
11:03:34.479: Out SSL V-MIDI Port 8 Sysex [f0 01 2e 0e f7]
11:03:34.582: Out SSL V-MIDI Port 9 Sysex [f0 01 2e 0e f7]
11:03:34.685: Out SSL V-MIDI Port 10 Sysex [f0 01 2e 0e f7]
11:03:34.788: Out SSL V-MIDI Port 11 Sysex [f0 01 2e 0e f7]
11:03:34.891: Out SSL V-MIDI Port 12 Sysex [f0 01 2e 0e f7]
11:03:34.994: Out Focusrite Thunderbolt MIDI Sysex [f0 01 2e 0e f7]
11:03:35.096: Out UM-ONE Sysex [f0 01 2e 0e f7]
11:03:35.117: In  UM-ONE Sysex [f0 01 2e 0f 0c 32 11 02 00 02 02 01 00 00 00 00 02 00 00 00 00 00 00 00 01 2f 00 01 01 f7]
```

The error I get is the following

`21:57:18: No DSI OB-6 could be detected - is it turned on?`

**Comment by christofmuc** [comment]
@Florent-H Awsome, that helped! It is indeed that the MIDI channel in MPE mode is reported as MIDI channel 0x11 = 17, which is unexpected. It is interpreted a) as Omni and b) as Invalid, because it is definitely not in the MIDI standard.

I will do a fix and release, probably will get around to do it on the weekend!

Thanks again for reporting this!

**Comment by Florent-H** [comment]
@christofmuc Amazing! Thanks a lot =)

**Comment by christofmuc** [comment]
@Florent-H Released with version 1.13.5

**Comment by Florent-H** [comment]
@christofmuc Awesome! Thanks for the new release. I will test it in a few days.
